### DISCUSSION TITLE
Latest Version Web-Boxedwine

### DISCUSSION THREAD

**Discussion by Klubuntu** [danoon2/Boxedwine#50] [Ideas]
**Hi, how to compile latest version Boxedwine (with wine 6.2/7) for Web ?**

**Comment by danoon2** [comment]
Sometime around Wine 5.12, they made some changes that break Boxedwine.  And in Wine 6/7, they made lots of changes to the windowing driver they use for Mac and X11.  I have my own version of this driver which improves performance a lot for Boxedwine, but it means I have to keep up with their changes.

I think I'm mostly caught up with the windowing changes  in my vulkan branch.  I'm currently working on the changes they made around Wine 5.12.  This change involves using shared memory to communicate between Wine processes.  For example, previously GetTickCount would ask Linux for the time.  Now wineserver will update shared memory with the correct tick count and each Wine process (games/apps) will read from shared memory to get the tick count.  Boxedwine did not fully implement shared memory, so now I'm adding that.   The current state is that it works for the slower versions of Boxedwine, but the fast version (binary translator) for ARMv8 and x64 don't.

I do have a build of Wine 7 I'm currently experimenting with.  With any luck, I should be able to release along with the changes I mention above in the next couple of months.

**Reply by Klubuntu** [reply]
or if it works, you will release a version for the web

**Comment by danoon2** [comment]
Is there something specific you are looking for from Wine 7?  Normally I tend to use the lowest version of Wine that works for web just because how much smaller it is.

Wine 1.7 is 31.5 MB
Wine 5.0 is 65.0 MB
Wine 7.0 is 77.4 MB

**Reply by Klubuntu** [reply]
I'm have a lot of space on hosting, and the latest version of wine would be useful for my project

**Reply by danoon2** [reply]
FYI, I'm still working on Wine 6/7 when I get time.  They required a lot of changes because the internal structures now use shared memory between clients and wineserver instead of pipe messages.
